README.md

```diff
@@ -1 +1,78 @@
 # rancher-desktop-networking
+`rancher-desktop-networking` is a simple layer 2 switch that allows connectivity between the VM and the host over `AF_VSOCK` written in Go using [gvisor's](https://github.com/google/gvisor) network stack and [gvisor-tap-vsock](https://github.com/google/gvisor). In addition to piping the ethernet frames, it also provides a DNS, DHCP services and dynamic port forwarding.
+
+Rancher Desktop Networking consists of three main components: host-switch, vm-switch, and network-setup. Below is a brief explanation of what each component does.
+
+## How it works
+```mermaid
```

Comment:
Please have a comment (at least, as a comment that isn't rendered) describing what the shapes mean — what are rectangles, diamonds, and circles?